File: Programmers/[81301] 숫자 문자열과 영단어/SUbbb/81301.java
import java.util.*;

class Solution {
    String[] table = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    public int solution(String s) {
        StringBuilder answer = new StringBuilder();
        Map<String, Integer> map = new HashMap<>();

        int idx = 0;
        for (String t : table) map.put(t, idx++);

        idx = 0;
        while(idx < s.length()) {
            char ch = s.charAt(idx);
            if (Character.isDigit(ch)) {
                answer.append(ch);
                idx++;
            } else {
                for (String key : map.keySet()) {
                    if (s.contains(key)) {
                        int start = s.indexOf(key);
                        if (start != idx) continue;
                        s = s.substring(0, start) + s.substring(start + key.length());
                        answer.append(map.get(key));
                        break;
                    }
                }
            }
        }

        return Integer.parseInt(answer.toString());
    }
}

File: Programmers/[81301] 숫자 문자열과 영단어/Johoseong/81301.py
def solution(s):
    words = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']

    arr = s
    for i in range(10):
        arr = arr.replace(words[i], str(i))
    
    return int(arr)

print(solution("one4seveneight"))